### ISSUE TITLE
Fluent rx names

### ISSUE THREAD

**Issue by carlosypunto** [ReactiveX/RxSwift#108]
Having spent from function composition to protocol extension some terms cease to be obvious. Like `subscribeTextOf` which it is a bit confusing. Now is the time since the 2.0 API is not final.
Open this thread for the native speakers can help to polish some confusing terms.

**Comment by bontoJR** [comment]
:+1: Perfect timing, actually the prefix issue is still open, @kzaher what do we do whit that as well?

**Comment by lfaoro** [comment]
There are only two hard things in Computer Science: cache invalidation and naming things.

**Comment by kzaher** [comment]
@bontoJR I think this is actually a separate issue.

We have issue of naming conventions for extensions for external components like `UIButton`, and how to name observable extensions to enable writing fluent code that binds to those components.

I'm not a native english speaker, so I am wondering how does

```
texts.subscribeTextOf(textField1)
```

```
allUsers
            .subscribe(tableView, withReactiveDataSource: dataSource)
            .addDisposableTo(disposeBag)
```

sound to native speakers.

**Comment by kzaher** [comment]
The reason why I'm using items for tableView as for collection view is because data sources for those 2 are the same, and only view is different, so I would ideally want that user to be able to just replace view type and type of outlet and the rest of the code should work the same.

I understand that `subscribeRowsOf` would probably be more correct, but I think `row` is part of `UITableView` legacy, and probably the correct thing would be for Apple to unify them somehow :)

Any thoughts on this would also be interesting ...

**Comment by orta** [comment]
anytime we do `subscribeItemsOf` it should really be `subscribeToItemsOf`

**Comment by DanielAsher** [comment]
how does the following read to you?

```
texts.subscribe(toTextOf: textField1)
allUsers.subscribe(tableView, to: dataSource, with: disposeBag)
```
how does code completion work with this?

**Comment by kzaher** [comment]
I like that `texts.subscribe(toTextOf: textField1)` format.

I think I've got an elegant solution, we'll make `textField1.text` a subject. And also make tableView.items a curried function that would take a data source and return an observer :)

In that way you'll be able to write:

```
texts.subscribe(textField1.text)
allUsers
     .subscribe(tableView.rx_items(dataSource))
     .addDisposableTo(disposeBag)
```

The good thing is that there aren't any new concepts introduced, the bad thing is it does have a double `))`, but it's kind of not that bad :)

**Comment by DanielAsher** [comment]
I think a studied decision needs to be made about the use of parameter names in signatures.
I haven't come to any conclusions myself, but I know they can assist readability, particularly outside of an IDE. If found to be the best solution I'd favour:
```
texts.subscribe(to: textField1.Text)
allUsers.subscribe(to: tableView.Items, with: dataSource, disposingWith: disposeBag)
```
I've included a suggestion for capitalized property fields to indicate, by convention, reactive properties.
I don't think there is a problem with ambiguity if the convention is quickly establish as there are no competing capitalization conventions on properties. But it certainly requires more thought.

**Comment by DanielAsher** [comment]
after further consideration, and using capitalization in real code, I understand capitalization won't work.

sorry for the noise.

**Comment by orta** [comment]
Better to have explored the possibility space @DanielAsher :+1:

**Comment by kzaher** [comment]
@DanielAsher I don't mind jiggling neurons by unexpected suggestions :+1:

**Comment by kzaher** [comment]
I've made a PR for this

https://github.com/ReactiveX/RxSwift/pull/119

Hope this makes sense :)

**Comment by DanielAsher** [comment]
Interesting! I'll check it out...

**Comment by bontoJR** [comment]
This seems to be fixed with PR #119.
